packages/x-components/src/x-modules/scroll/components/__tests__/window-scroll.spec.ts

```diff
@@ -522,42 +601,63 @@ describe('testing Main Scroll Component', () => {
         durationMs: 200
       });
 
+      expect(listenerAlmostReachedScrollEnd).toHaveBeenCalledTimes(2);
+      expect(listenerReachedScrollEnd).toHaveBeenCalledTimes(3);
+      expect(listenerAlmostReachedScrollEnd).toHaveBeenNthCalledWith(2, {
+        eventPayload: false,
+        metadata: {
+          moduleName: 'scroll',
+          target: document.body,
+          id: 'main-scroll'
+        }
+      });
+      expect(listenerReachedScrollEnd).toHaveBeenNthCalledWith(3, {
+        eventPayload: false,
+        metadata: {
+          moduleName: 'scroll',
+          target: document.body,
+          id: 'main-scroll'
+        }
+      });
+
       await scrollBody({
         to: 600,
         durationMs: 200
       });
 
-      expect(listenerAlmostReachedScrollEnd).toHaveBeenNthCalledWith(2, {
-        eventPayload: 0,
+      expect(listenerAlmostReachedScrollEnd).toHaveBeenCalledTimes(3);
+      expect(listenerReachedScrollEnd).toHaveBeenCalledTimes(4);
+      expect(listenerAlmostReachedScrollEnd).toHaveBeenNthCalledWith(3, {
+        eventPayload: true,
         metadata: {
-          moduleName: null, // no module registered for this base component
+          moduleName: 'scroll',
           target: document.body,
           id: 'main-scroll'
         }
       });
-      expect(listenerReachedScrollEnd).toHaveBeenNthCalledWith(2, {
-        eventPayload: undefined,
+      expect(listenerReachedScrollEnd).toHaveBeenNthCalledWith(4, {
+        eventPayload: true,
         metadata: {
-          moduleName: null, // no module registered for this base component
+          moduleName: 'scroll',
           target: document.body,
           id: 'main-scroll'
         }
       });
     });
 
     it('does not trigger event when scrolling in document or other element', async () => {
-      const { wrapperContainer, scrollHtml, scrollContent } = renderBaseMainScroll({
+      const { wrapper, scrollHtml, scrollContent } = await renderWindowScroll({
```

Comment:
Should this test be removed? I feel like it is testing a native browser behavior (the scroll event does not bubble), and it is doing so by mocking all the JSDOM elements in the test with random values....

Reply:
I like the validation that our scrolls are not scrolling, for me is ok to leave it but I'll buy you if you prefer to delete it.

Reply:
We can delete it if it doesn't provide any value. We can check it later after the whole module's rework or in another pr, as I'm going to merge this one.